Roadmap preview: July 2025 (#29757)
fleetdm/fleet: articles/roadmap-preview-july-2025.md

# Roadmap preview, July 2025

<div purpose="embedded-content">
   <iframe src="TODO" allowfullscreen></iframe>
</div>

The Fleet roadmap is set for summer 2025. Watch the video above for a walkthrough, or continue reading for the highlights.

In the next 3 months, Fleet will ship...

- 🍏 Account-based user enrollment for personal Apple devices (BYOD)
- 🤖 Manage BYOD Android: enforce OS settings and manage software in work profiles
- 📄 Deploy certificates for Wi-Fi and VPN on personal mobile devices (iOS/iPadOS, Android) and company-owned Windows, & Linux computers

> - 👨‍💻 100 Fleet-maintained apps for macOS & Windows (50 each)

Author: ```suggestion
- 👨‍💻 More Windows Fleet-maintained apps: Microsoft Office (Word, Excel, Teams), Google Chrome, Slack, and more.
```

- 📦 Different apps, OS settings, and queries for different employees based on department
- 👔 Customize first-time Windows and Linux setup

> - 💻 Inject foreign host vitals from your identity provider (IdP) and mask secrets (i.e. API tokens) in scripts and configuration profiles

Author: ```suggestion
- 💻 Inject foreign host vitals from your identity provider (IdP) and variables in scripts and configuration profiles
```

- 🧪 Deploy your pre-release iOS apps to the entire company
- ⏰ Schedule scripts to run once, at a specific time, or on a recurring schedule

Big opportunities that Fleet is building towards in the near future (next 180 days):

> - 📊 Vulnerability (CVE) graphs/visualizations

Author: ```suggestion
- 📊 Vulnerability burn-down chart (open CVEs over time)
```

> - 🔊 Network scanning (via Nmap)

Author: ```suggestion
- 🔊 Find missing assets and their vulnerabilities (via Nmap)
```

> - 📉 Report time-to-fix (aka "timeframes") for vulnerabilities

Author: ```suggestion
```

# Roadmap preview, July 2025

<div purpose="embedded-content">
   <iframe src="TODO" allowfullscreen></iframe>
</div>

The Fleet roadmap is set for summer 2025. Watch the video above for a walkthrough, or continue reading for the highlights.

In the next 3 months, Fleet will ship...

For Security Engineers:
- 📊 Vulnerability (CVE) graphs/visualizations
- 🔍 Complex host filters (any host vital)
- 📄 Deploy certificates for Wi-Fi and VPN on Windows and Linux computers

For IT Admins:
- 🍏 Account-based user enrollment for personal Apple devices (BYOD)

> - 👨‍💻 Update 100 Fleet-maintained apps for macOS & Windows (50 each)

Author: ```suggestion
- 👨‍💻 100 Fleet-maintained apps for macOS & Windows (50 each)
```

- 📦 Different apps, OS settings, and queries for different employees based on department
- 👔 Custom new Windows and Linux setup
- 💻 Host vitals as variables in scripts and configuration profiles
- 🤖 Configuration profiles, OS updates, and software management for BYOD Android

> - 🧪 Deploy your in-house iOS apps so the entire company can test

Author: ```suggestion
- 🧪 Deploy your in-house iOS apps so the entire company can test
- ⏰ Schedule scripts to run once, at a specific time, or on a recurring schedule
```

Big opportunities that Fleet is building towards in the near future (next 180 days):
- 🤖 Auto-complete osquery queries (SQL), Bash scripts, and PowerShell scripts

> - 🔊 Network scanning

Author: ```suggestion
- 🔊 Network scanning (Nmap)
```

> - 🔊 Network scanning (Nmap)

Author: ```suggestion
- 🔊 Network scanning (via Nmap)
```

# Roadmap preview, July 2025

<div purpose="embedded-content">
   <iframe src="TODO" allowfullscreen></iframe>
</div>

The Fleet roadmap is set for summer 2025. Watch the video above for a walkthrough, or continue reading for the highlights.

In the next 3 months, Fleet will ship...

For Security Engineers:
- 📊 Vulnerability (CVE) graphs/visualizations
- 🔍 Complex host filters (any host vital)
- 📄 Deploy certificates for Wi-Fi and VPN on personal mobile devices (iOS/iPadOS, Android) and company-owned Windows, & Linux computers

For IT Admins:
- 🍏 Account-based user enrollment for personal Apple devices (BYOD)
- 👨‍💻 100 Fleet-maintained apps for macOS & Windows (50 each)
- 📦 Different apps, OS settings, and queries for different employees based on department
- 👔 Custom new Windows and Linux setup

> - 💻 Inject foreign host vitals from your identity provider (IdP) in scripts and configuration profiles

Author: ```suggestion
- 💻 Inject foreign host vitals from your identity provider (IdP) and mask secrets (i.e. API tokens) in scripts and configuration profiles
```

- 🤖 Configuration profiles, OS updates, and software management for BYOD Android
- 🧪 Deploy your in-house iOS apps so the entire company can test
- ⏰ Schedule scripts to run once, at a specific time, or on a recurring schedule

Big opportunities that Fleet is building towards in the near future (next 180 days):

> - 🤖 Auto-complete osquery queries (SQL), Bash scripts, and PowerShell scripts

Author: ```suggestion
- 📊 Vulnerability (CVE) graphs/visualizations
```
